### PR TITLE
Add script to create a Federalist environment

### PR DESCRIPTION
**Pull request by davemcorwin**
## Changes proposed in this pull request:
- Adds a script to setup an entire Federalist environment

## security considerations
Manually verified all gitleaks findings, no leaks. An issue has been created for preventing false positives when using environment variable.

### CODE REVIEW

**Comment by svenaas** [comment]
@davemcorwin this is awesome.

I don't know of any way to test it, though, and I think @apburnes would be the Most Valuable Reviewer here.

File: scripts/create-instance.sh [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+EOF
+)"
+
+# Grant permissions in the sites space
+cf set-space-role "${api_user_username}" "${cf_org}" "${cf_sites_space}" "SpaceDeveloper"
+
+# Temporarily create a service key to get the bucket name
+tmp_s3_service_key_name="${service_name_s3}-key-tmp"
+cf create-service-key "${service_name_s3}" "${tmp_s3_service_key_name}"
+shared_bucket=$(cf service-key "${service_name_s3}" "${tmp_s3_service_key_name}" | tail -n +2 | jq -r .bucket)
```
Comment (apburnes) [inline scripts/create-instance.sh:243]: Not sure we need a shared bucket in new space creation.

Comment (apburnes) [reply scripts/create-instance.sh:243]: Just legacy to deploy the proxy?

File: scripts/create-instance.sh [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+echo
+echo "-  In the Concourse jumpbox, please run the following:"
+echo "     credhub set -n /concourse/pages/${cf_space}-cf-password -t value -v ${ci_user_password}"
+echo "     credhub set -n /concourse/pages/${cf_space}-cf-username -t value -v ${ci_user_username}"
+echo
+echo "-  Add any additional humans:"
+echo "     cf set-space-role <username@gsa.gov> ${cf_org} ${cf_space} SpaceDeveloper"
+echo "     cf set-space-role <username@gsa.gov> ${cf_org} ${cf_sites_space} SpaceDeveloper"
+echo
+echo "You may now deploy the pages and applications:"
```
Comment (apburnes) [inline scripts/create-instance.sh:384]: Be more explicit about kicking off a build in CI?

File: scripts/create-instance.sh (hunk 1 above)
Comment (davemcorwin) [reply scripts/create-instance.sh:243]: yeah, we still have a few stranglers, we can remove the shared bucket as a separate issue.

File: scripts/create-instance.sh (hunk 2 above)
Comment (davemcorwin) [reply scripts/create-instance.sh:384]: What do you mean?

Comment (apburnes) [reply scripts/create-instance.sh:384]: If we want any more documentation on using Concourse or Circle to kick off a deploy. But it's not like we're going to use this that offten.

**Review by apburnes — APPROVED** [review]

**Pull request merged**